pkg/clusteragent/autoscaling/workload/controller_horizontal.go

```diff
@@ -55,15 +57,24 @@ func newHorizontalReconciler(clock clock.Clock, eventRecorder record.EventRecord
 }
 
 func (hr *horizontalController) sync(ctx context.Context, podAutoscaler *datadoghq.DatadogPodAutoscaler, autoscalerInternal *model.PodAutoscalerInternal) (autoscaling.ProcessResult, error) {
+	// If we have no Spec, nothing to do
+	if autoscalerInternal.Spec() == nil {
+		return autoscaling.NoRequeue, nil
```

Comment:
It would be a silent failure. Is it something we expect ? Should we return an error and call `UpdateFromHorizontalAction` ?